modules/compiler/utils/source/replace_atomic_funcs_pass.cpp

```diff
@@ -225,20 +363,21 @@ bool RunOnInstruction(CallInst &call) {
       auto op1 =
           (1 == call.arg_size()) ? builder.getInt32(1) : call.getArgOperand(1);
 
-      if (call.getType()->isFloatTy()) {
-        auto ptr = builder.getInt32Ty()->getPointerTo(
-            op0->getType()->getPointerAddressSpace());
+      if (call.getType()->isFloatingPointTy()) {
+        auto ptr = builder.getIntNTy(call.getType()->getPrimitiveSizeInBits())
```

Comment:
This isn't strictly necessary as we don't support typed pointers, but I don't mind it either way. This is arguably more versatile.

Reply:
True, in opaque pointer mode (the only mode still in use) this just creates a bitcast from `ptr` to `ptr` (except in the right address space). We already did so for `float`, this change makes it so that `double` behaves the same as `float`, but we could take this out for both `float` and `double` now that it is no longer needed. We have tickets to remove specific bits of leftover support for older LLVM versions, I'll create a similar ticket to go over the remaining typed pointer support and take it out.